<a href="https://colab.research.google.com/github/rhks13/Closure-Prediction/blob/main/%EC%83%81%EA%B6%8C%EC%BD%94%EB%93%9C%EC%97%B0%EA%B2%B0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# 드라이브 마운트
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
import numpy as np
import os
from glob import glob
import pickle

In [ ]:
상권 = pd.read_excel("/content/drive/MyDrive/통계 빅데이터 공모전/data/상권 특성/상권리스트.xlsx")

### **geometry의 x/y 최대최소 이용**

In [ ]:
trdar = 상권.copy()
trdar.geometry

0       192008.56 438553.17, 191987.74 438544.51, 1919...
1       206296.88 459321.71, 206238.71 459299.03, 2062...
2       192443.04 439028.23, 192507.94 438996.04000000...
3       191390.84 437878.79000000004, 191371.65 437875...
4       210354.67 445450.29000000004, 210314.56 445371...
                              ...                        
1666    201553.986325298 452217.686202384, 201565.1128...
1667    198035.74942062798 452283.91062634904, 198019....
1668    208228.91417088 445674.946959898, 208197.64944...
1669    205328.897308437 446048.922964627, 205339.5607...
1670    199226.204779015 447867.577119885, 199222.7606...
Name: geometry, Length: 1671, dtype: object

In [ ]:
geom=trdar['geometry'].str.split(' ')

In [ ]:
geom

0       [192008.56, 438553.17,, 191987.74, 438544.51,,...
1       [206296.88, 459321.71,, 206238.71, 459299.03,,...
2       [192443.04, 439028.23,, 192507.94, 438996.0400...
3       [191390.84, 437878.79000000004,, 191371.65, 43...
4       [210354.67, 445450.29000000004,, 210314.56, 44...
                              ...                        
1666    [201553.986325298, 452217.686202384,, 201565.1...
1667    [198035.74942062798, 452283.91062634904,, 1980...
1668    [208228.91417088, 445674.946959898,, 208197.64...
1669    [205328.897308437, 446048.922964627,, 205339.5...
1670    [199226.204779015, 447867.577119885,, 199222.7...
Name: geometry, Length: 1671, dtype: object

In [ ]:
l1=[]

for j in range(len(geom[0])):
  l1.append(geom[0][j].split(','))
l2 = pd.DataFrame(l1).iloc[:,0]
l2 = l2.astype(float)
y_max = l2[1::2].max()
y_min = l2[1::2].min()
x_max = l2[::2].max()
x_min = l2[::2].min()

In [ ]:
#초기 데이터프레임 생성
ran = [x_min, x_max, y_min, y_max]
df11 = pd.DataFrame(ran).T
df11

0          1          2          3
0  191626.74  192242.12  438544.51  439081.47

In [ ]:
for i in range(1,len(geom)): 
  l1=[]
  
  for j in range(len(geom[i])):
    l1.append(geom[i][j].split(','))
  l2 = pd.DataFrame(l1).iloc[:,0]
  try:
    l2 = l2.astype(float) 
  except:
     pass
  y_max = l2[1::2].max()
  y_min = l2[1::2].min()
  x_max = l2[::2].max()
  x_min = l2[::2].min()
  ran = [x_min, x_max, y_min, y_max]
  df22 = pd.DataFrame(ran).T
  df11 = df11.append(df22)


In [ ]:
len(geom)

1671

In [ ]:
df11.index=trdar['TRDAR_NO']

In [ ]:
df11['num']=range(len(df11))

In [ ]:
df11[0]=df11[0].astype(float)
df11[1]=df11[1].astype(float)
df11[2]=df11[2].astype(float)
df11[3]=df11[3].astype(float)

In [ ]:
df11

0              1              2              3   num
TRDAR_NO                                                                  
2110753   191626.740000  192242.120000  438544.510000  439081.470000     0
2110427   206170.860000  206358.740000  459299.030000  459563.600000     1
2110755   192041.700000  192557.920000  438960.280000  439178.990000     2
2110739   191108.060000  191470.440000  437849.160000  438178.930000     3
2111017   210071.420000  210372.460000  445371.860000  445659.890000     4
...                 ...            ...            ...            ...   ...
1001493   200377.852920  201590.605471  451670.967652  452246.400447  1666
1001494   198013.331674  201383.463880  452065.426802  452611.339129  1667
1001495   208197.649441  211576.892042  445156.444683  447387.110211  1668
1001496   205039.806949  205560.632953  445381.595583  446048.922965  1669
1001491   198944.696026  200239.739917  447860.055959  448825.637470  1670

[1671 rows x 5 columns]

In [ ]:
with open('/content/drive/MyDrive/통계 빅데이터 공모전/data/업체 데이터/데이터최종원본.pickle', 'rb') as f:
    main_df = pickle.load(f)  

In [ ]:
main_data = main_df.copy()
main_data

번호 개방서비스명    개방서비스아이디  개방자치단체코드                    관리번호     인허가일자  \
0       37768  일반음식점  07_24_04_P   3010000  3010000-101-2017-00001  20170102   
1       51499  일반음식점  07_24_04_P   3020000  3020000-101-2017-00001  20170102   
2       54896  일반음식점  07_24_04_P   3030000  3030000-101-2017-00001  20170102   
3       54897  일반음식점  07_24_04_P   3030000  3030000-101-2017-00002  20170102   
4       85257  일반음식점  07_24_04_P   3050000  3050000-101-2017-00003  20170102   
...       ...    ...         ...       ...                     ...       ...   
67302  363189  일반음식점  07_24_04_P   3210000  3210000-101-2021-00614  20211231   
67303  419761  일반음식점  07_24_04_P   3220000  3220000-101-2021-01572  20211231   
67304  419762  일반음식점  07_24_04_P   3220000  3220000-101-2021-01573  20211231   
67305  433467  일반음식점  07_24_04_P   3230000  3230000-101-2021-00922  20211231   
67306  457441  일반음식점  07_24_04_P   3240000  3240000-101-2021-00484  20211231   

       인허가취소일자  영업상태구분코드  영업상태명  상세영업상태코드  ... 영업장주변구분명  등급구분명  급수시설구분명  \
0          NaN         3     폐업         2  ...      NaN    NaN      NaN   
1          NaN         3     폐업         2  ...      NaN    NaN      NaN   
2          NaN         1  영업/정상         1  ...      NaN    NaN      NaN   
3          NaN         1  영업/정상         1  ...      NaN    NaN      NaN   
4          NaN         3     폐업         2  ...      NaN    NaN      NaN   
...        ...       ...    ...       ...  ...      ...    ...      ...   
67302      NaN         1  영업/정상         1  ...      NaN    NaN      NaN   
67303      NaN         1  영업/정상         1  ...      NaN    NaN      NaN   
67304      NaN         1  영업/정상         1  ...      NaN    NaN      NaN   
67305      NaN         1  영업/정상         1  ...      NaN    NaN      NaN   
67306      NaN         1  영업/정상         1  ...      NaN    NaN      NaN   

       다중이용업소여부   시설총규모         추정임대료 가맹여부   업종분류 인허가년도 인허가분기  
0             N   29.88  9.077544e+08    0  한식음식점  2017     1  
1             N   72.99  6.266337e+08    0     기타  2017     1  
2             N   36.61  2.794441e+08    1  한식음식점  2017     1  
3             N  126.46  8.013770e+08    0  한식음식점  2017     1  
4             N   47.94  3.716405e+08    0  한식음식점  2017     1  
...         ...     ...           ...  ...    ...   ...   ...  
67302         N   32.09           NaN    1  분식전문점  2021     4  
67303         Y  163.11  3.833085e+09    0  양식음식점  2021     4  
67304         Y  342.66  4.814373e+09    0  한식음식점  2021     4  
67305         N   27.40           NaN    0  분식전문점  2021     4  
67306         N   38.60  1.985198e+08    0     기타  2021     4  

[67307 rows x 41 columns]

In [ ]:
main_data['상권코드'] = 0

In [ ]:
main_data['번호']=range(len(main_data))

In [ ]:
main_data

번호 개방서비스명    개방서비스아이디  개방자치단체코드                    관리번호     인허가일자  \
0          0  일반음식점  07_24_04_P   3010000  3010000-101-2017-00001  20170102   
1          1  일반음식점  07_24_04_P   3020000  3020000-101-2017-00001  20170102   
2          2  일반음식점  07_24_04_P   3030000  3030000-101-2017-00001  20170102   
3          3  일반음식점  07_24_04_P   3030000  3030000-101-2017-00002  20170102   
4          4  일반음식점  07_24_04_P   3050000  3050000-101-2017-00003  20170102   
...      ...    ...         ...       ...                     ...       ...   
67302  67302  일반음식점  07_24_04_P   3210000  3210000-101-2021-00614  20211231   
67303  67303  일반음식점  07_24_04_P   3220000  3220000-101-2021-01572  20211231   
67304  67304  일반음식점  07_24_04_P   3220000  3220000-101-2021-01573  20211231   
67305  67305  일반음식점  07_24_04_P   3230000  3230000-101-2021-00922  20211231   
67306  67306  일반음식점  07_24_04_P   3240000  3240000-101-2021-00484  20211231   

       인허가취소일자  영업상태구분코드  영업상태명  상세영업상태코드  ... 등급구분명  급수시설구분명  다중이용업소여부  \
0          NaN         3     폐업         2  ...   NaN      NaN         N   
1          NaN         3     폐업         2  ...   NaN      NaN         N   
2          NaN         1  영업/정상         1  ...   NaN      NaN         N   
3          NaN         1  영업/정상         1  ...   NaN      NaN         N   
4          NaN         3     폐업         2  ...   NaN      NaN         N   
...        ...       ...    ...       ...  ...   ...      ...       ...   
67302      NaN         1  영업/정상         1  ...   NaN      NaN         N   
67303      NaN         1  영업/정상         1  ...   NaN      NaN         Y   
67304      NaN         1  영업/정상         1  ...   NaN      NaN         Y   
67305      NaN         1  영업/정상         1  ...   NaN      NaN         N   
67306      NaN         1  영업/정상         1  ...   NaN      NaN         N   

        시설총규모         추정임대료 가맹여부   업종분류  인허가년도 인허가분기 상권코드  
0       29.88  9.077544e+08    0  한식음식점   2017     1    0  
1       72.99  6.266337e+08    0     기타   2017     1    0  
2       36.61  2.794441e+08    1  한식음식점   2017     1    0  
3      126.46  8.013770e+08    0  한식음식점   2017     1    0  
4       47.94  3.716405e+08    0  한식음식점   2017     1    0  
...       ...           ...  ...    ...    ...   ...  ...  
67302   32.09           NaN    1  분식전문점   2021     4    0  
67303  163.11  3.833085e+09    0  양식음식점   2021     4    0  
67304  342.66  4.814373e+09    0  한식음식점   2021     4    0  
67305   27.40           NaN    0  분식전문점   2021     4    0  
67306   38.60  1.985198e+08    0     기타   2021     4    0  

[67307 rows x 42 columns]

In [ ]:

def trdar(i):
  try:
    x=main_data['좌표정보(X)'].astype(float)
    y=main_data['좌표정보(Y)'].astype(float)
    return np.where((x>=df11.iloc[i,0])&(x<=df11.iloc[i,1])&(y>=df11.iloc[i,2])&(y<=df11.iloc[i,3]))
  except:
    pass

In [ ]:
df11['상권'] = df11['num'].map(lambda x : trdar(x))

In [ ]:
main_data.describe().T

count          mean           std           min           25%  \
번호        67307.0  3.365300e+04  1.943000e+04  0.000000e+00  1.682650e+04   
개방자치단체코드  67307.0  3.135827e+06  7.373946e+04  3.000000e+06  3.070000e+06   
인허가일자     67307.0  2.019118e+07  1.408879e+04  2.017010e+07  2.018043e+07   
인허가취소일자       0.0           NaN           NaN           NaN           NaN   
영업상태구분코드  67307.0  1.608198e+00  9.200565e-01  1.000000e+00  1.000000e+00   
상세영업상태코드  67307.0  1.304099e+00  4.600283e-01  1.000000e+00  1.000000e+00   
폐업일자      20468.0  2.020113e+07  1.264515e+04  2.017012e+07  2.019082e+07   
휴업시작일자        0.0           NaN           NaN           NaN           NaN   
휴업종료일자        0.0           NaN           NaN           NaN           NaN   
재개업일자         0.0           NaN           NaN           NaN           NaN   
소재지우편번호   67290.0  1.372575e+05  1.383596e+04  1.000110e+05  1.318120e+05   
도로명우편번호   67006.0  5.278211e+03  2.102466e+03  1.000000e+03  3.787000e+03   
최종수정시점    67307.0  2.020101e+13  1.302706e+10  2.017010e+13  2.019090e+13   
좌표정보(X)   66203.0  1.990143e+05  7.351135e+03  1.779157e+05  1.929610e+05   
좌표정보(Y)   66203.0  4.490393e+05  5.255851e+03  4.368888e+05  4.448367e+05   
남성종사자수    16506.0  1.029929e-02  2.644491e-01  0.000000e+00  0.000000e+00   
여성종사자수    16501.0  9.878189e-03  2.380972e-01  0.000000e+00  0.000000e+00   
시설총규모     67307.0  7.225117e+01  1.097538e+02  0.000000e+00  2.886000e+01   
추정임대료     51035.0  7.122038e+08  1.961944e+09  0.000000e+00  1.517958e+08   
가맹여부      67307.0  1.816602e-01  3.855671e-01  0.000000e+00  0.000000e+00   
인허가년도     67307.0  2.019051e+03  1.409012e+00  2.017000e+03  2.018000e+03   
인허가분기     67307.0  2.524849e+00  1.097708e+00  1.000000e+00  2.000000e+00   
상권코드      67307.0  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   

                   50%           75%           max  
번호        3.365300e+04  5.047950e+04  6.730600e+04  
개방자치단체코드  3.140000e+06  3.210000e+06  3.240000e+06  
인허가일자     2.019073e+07  2.020102e+07  2.021123e+07  
인허가취소일자            NaN           NaN           NaN  
영업상태구분코드  1.000000e+00  3.000000e+00  3.000000e+00  
상세영업상태코드  1.000000e+00  2.000000e+00  2.000000e+00  
폐업일자      2.020093e+07  2.021073e+07  2.022053e+07  
휴업시작일자             NaN           NaN           NaN  
휴업종료일자             NaN           NaN           NaN  
재개업일자              NaN           NaN           NaN  
소재지우편번호   1.378030e+05  1.507060e+05  4.118060e+05  
도로명우편번호   5.331000e+03  7.008000e+03  1.038300e+04  
최종수정시점    2.020103e+13  2.021080e+13  2.022053e+13  
좌표정보(X)   1.999859e+05  2.044830e+05  2.160294e+05  
좌표정보(Y)   4.487996e+05  4.518840e+05  4.652729e+05  
남성종사자수    0.000000e+00  0.000000e+00  2.500000e+01  
여성종사자수    0.000000e+00  0.000000e+00  2.500000e+01  
시설총규모     4.724000e+01  8.250000e+01  7.100000e+03  
추정임대료     2.955000e+08  6.517987e+08  1.479602e+11  
가맹여부      0.000000e+00  0.000000e+00  1.000000e+00  
인허가년도     2.019000e+03  2.020000e+03  2.021000e+03  
인허가분기     3.000000e+00  4.000000e+00  4.000000e+00  
상권코드      0.000000e+00  0.000000e+00  0.000000e+00

In [ ]:
1#빈 리스트에 모두 추가하는 방식
f_list=[]
for i in range(len(df11)):
  try:
    for j in range(len(df11['상권'].iloc[i][0])):
       f_list.append(df11['상권'].iloc[i][0][j])
  except:
    pass  

In [ ]:
len(f_list)

84189

In [ ]:
len(set(f_list))

53732

In [ ]:
df11

0              1              2              3   num  \
TRDAR_NO                                                                     
2110753   191626.740000  192242.120000  438544.510000  439081.470000     0   
2110427   206170.860000  206358.740000  459299.030000  459563.600000     1   
2110755   192041.700000  192557.920000  438960.280000  439178.990000     2   
2110739   191108.060000  191470.440000  437849.160000  438178.930000     3   
2111017   210071.420000  210372.460000  445371.860000  445659.890000     4   
...                 ...            ...            ...            ...   ...   
1001493   200377.852920  201590.605471  451670.967652  452246.400447  1666   
1001494   198013.331674  201383.463880  452065.426802  452611.339129  1667   
1001495   208197.649441  211576.892042  445156.444683  447387.110211  1668   
1001496   205039.806949  205560.632953  445381.595583  446048.922965  1669   
1001491   198944.696026  200239.739917  447860.055959  448825.637470  1670   

                                                         상권  
TRDAR_NO                                                     
2110753   ([571, 828, 1105, 2846, 3259, 3296, 3796, 5821...  
2110427                      ([9900, 16498, 16499, 33565],)  
2110755                                               ([],)  
2110739   ([77, 395, 3682, 11849, 14331, 20856, 29716, 3...  
2111017   ([43077, 43297, 46648, 47489, 47809, 48334, 48...  
...                                                     ...  
1001493   ([0, 82, 728, 1346, 1348, 1378, 1865, 2393, 23...  
1001494   ([130, 188, 232, 256, 382, 383, 463, 600, 601,...  
1001495   ([42212, 42226, 42227, 42253, 42276, 42280, 42...  
1001496   ([42181, 42208, 42410, 42596, 42634, 43007, 43...  
1001491   ([65, 174, 175, 208, 262, 263, 304, 308, 385, ...  

[1671 rows x 6 columns]

In [ ]:
df11

0              1              2              3   num  \
TRDAR_NO                                                                     
2110753   191626.740000  192242.120000  438544.510000  439081.470000     0   
2110427   206170.860000  206358.740000  459299.030000  459563.600000     1   
2110755   192041.700000  192557.920000  438960.280000  439178.990000     2   
2110739   191108.060000  191470.440000  437849.160000  438178.930000     3   
2111017   210071.420000  210372.460000  445371.860000  445659.890000     4   
...                 ...            ...            ...            ...   ...   
1001493   200377.852920  201590.605471  451670.967652  452246.400447  1666   
1001494   198013.331674  201383.463880  452065.426802  452611.339129  1667   
1001495   208197.649441  211576.892042  445156.444683  447387.110211  1668   
1001496   205039.806949  205560.632953  445381.595583  446048.922965  1669   
1001491   198944.696026  200239.739917  447860.055959  448825.637470  1670   

                                                         상권  
TRDAR_NO                                                     
2110753   ([571, 828, 1105, 2846, 3259, 3296, 3796, 5821...  
2110427                      ([9900, 16498, 16499, 33565],)  
2110755                                               ([],)  
2110739   ([77, 395, 3682, 11849, 14331, 20856, 29716, 3...  
2111017   ([43077, 43297, 46648, 47489, 47809, 48334, 48...  
...                                                     ...  
1001493   ([0, 82, 728, 1346, 1348, 1378, 1865, 2393, 23...  
1001494   ([130, 188, 232, 256, 382, 383, 463, 600, 601,...  
1001495   ([42212, 42226, 42227, 42253, 42276, 42280, 42...  
1001496   ([42181, 42208, 42410, 42596, 42634, 43007, 43...  
1001491   ([65, 174, 175, 208, 262, 263, 304, 308, 385, ...  

[1671 rows x 6 columns]

## **상권리스트의 x/y 좌표 이용하기**

In [ ]:
x_range=(df11.iloc[:,1]-df11.iloc[:,0]).mean()

In [ ]:
y_range=(df11.iloc[:,3]-df11.iloc[:,2]).mean()

In [ ]:
x_std=(df11.iloc[:,1]-df11.iloc[:,0]).std()
y_std=(df11.iloc[:,3]-df11.iloc[:,2]).std()

In [ ]:
상권

Unnamed: 0  STDR_YM_CD TRDAR_SE_C TRDAR_SE_1  TRDAR_NO      TRDAR_NM  \
0              0      202112          A       골목상권   2110753   시흥동 은행나무사거리   
1              1      202112          A       골목상권   2110427    노원 을지대학교병원   
2              2      202112          A       골목상권   2110755      동일여자고등학교   
3              3      202112          A       골목상권   2110739    시흥3동 우편취급국   
4              4      202112          A       골목상권   2111017      송파나루역 3번   
...          ...         ...        ...        ...       ...           ...   
1666        1666      202112          U       관광특구   1001493  동대문패션타운 관광특구   
1667        1667      202112          U       관광특구   1001494    종로·청계 관광특구   
1668        1668      202112          U       관광특구   1001495       잠실 관광특구   
1669        1669      202112          U       관광특구   1001496   강남 마이스 관광특구   
1670        1670      202112          U       관광특구   1001491      이태원 관광특구   

      XCNTS_VALU  YDNTS_VALU  SIGNGU_CD  ADSTRD_CD  \
0         191934      438813      11545   11545710   
1         206265      459431      11350   11350611   
2         192300      439070      11545   11545680   
3         191289      438014      11545   11545690   
4         210222      445516      11710   11710562   
...          ...         ...        ...        ...   
1666      200974      451975      11140   11140590   
1667      199768      452273      11110   11110615   
1668      210191      446339      11710   11710566   
1669      205310      445727      11680   11680580   
1670      199503      448326      11170   11170650   

                                               geometry  
0     192008.56 438553.17, 191987.74 438544.51, 1919...  
1     206296.88 459321.71, 206238.71 459299.03, 2062...  
2     192443.04 439028.23, 192507.94 438996.04000000...  
3     191390.84 437878.79000000004, 191371.65 437875...  
4     210354.67 445450.29000000004, 210314.56 445371...  
...                                                 ...  
1666  201553.986325298 452217.686202384, 201565.1128...  
1667  198035.74942062798 452283.91062634904, 198019....  
1668  208228.91417088 445674.946959898, 208197.64944...  
1669  205328.897308437 446048.922964627, 205339.5607...  
1670  199226.204779015 447867.577119885, 199222.7606...  

[1671 rows x 11 columns]

In [ ]:
rang=pd.DataFrame({'xmin': 상권['XCNTS_VALU']-x_std})

In [ ]:
rang['xmax'] = 상권['XCNTS_VALU']+x_std
rang['ymin'] = 상권['YDNTS_VALU']-x_std
rang['ymax'] = 상권['YDNTS_VALU']+y_std
rang['num'] = range(len(rang['ymax']))

In [ ]:

def trdar(i):
  try:
    x=main_data['좌표정보(X)'].astype(float)
    y=main_data['좌표정보(Y)'].astype(float)
    return np.where((x>=rang.iloc[i,0])&(x<=rang.iloc[i,1])&(y>=rang.iloc[i,2])&(y<=rang.iloc[i,3]))
  except:
    pass

In [ ]:
rang['상권'] = rang['num'].map(lambda x : trdar(x))

In [ ]:
rang

xmin           xmax           ymin           ymax   num  \
0     191684.682928  192183.317072  438563.682928  439047.233954     0   
1     206015.682928  206514.317072  459181.682928  459665.233954     1   
2     192050.682928  192549.317072  438820.682928  439304.233954     2   
3     191039.682928  191538.317072  437764.682928  438248.233954     3   
4     209972.682928  210471.317072  445266.682928  445750.233954     4   
...             ...            ...            ...            ...   ...   
1666  200724.682928  201223.317072  451725.682928  452209.233954  1666   
1667  199518.682928  200017.317072  452023.682928  452507.233954  1667   
1668  209941.682928  210440.317072  446089.682928  446573.233954  1668   
1669  205060.682928  205559.317072  445477.682928  445961.233954  1669   
1670  199253.682928  199752.317072  448076.682928  448560.233954  1670   

                                                     상권  
0     ([571, 828, 1105, 2846, 3259, 3796, 5821, 6632...  
1     ([784, 904, 1556, 3980, 4681, 4918, 6830, 9900...  
2                                       ([2846, 8541],)  
3     ([77, 395, 573, 2332, 2333, 3682, 9069, 10195,...  
4     ([43077, 43297, 43925, 44419, 44452, 44894, 45...  
...                                                 ...  
1666  ([2393, 3592, 3927, 4378, 4455, 4456, 4488, 47...  
1667  ([6175, 7696, 8700, 10619, 11854, 11881, 12107...  
1668  ([42316, 44513, 51172, 51314, 54865, 55187, 55...  
1669  ([42181, 42208, 42410, 42596, 42634, 43007, 43...  
1670  ([5477, 5791, 6567, 7246, 9855, 10371, 11296, ...  

[1671 rows x 6 columns]

In [ ]:
1#빈 리스트에 모두 추가하는 방식
f_list=[]
for i in range(len(rang)):
  try:
    for j in range(len(rang['상권'].iloc[i][0])):
       f_list.append(rang['상권'].iloc[i][0][j])
  except:
    pass  

In [ ]:
len(f_list)

87725

In [ ]:
len(set(f_list))

51019

In [ ]:
f_list

[]

geometry의 2표준편차로 했을때 -> 87725/ 51019

geometry의 minmax평균으로 했을때 -> 66757/ 44750

geometry의 minmax 2/3로 나누었을때 -> 29964 / 25511

## **원으로 해보기**

In [ ]:
상권

Unnamed: 0  STDR_YM_CD TRDAR_SE_C TRDAR_SE_1  TRDAR_NO      TRDAR_NM  \
0              0      202112          A       골목상권   2110753   시흥동 은행나무사거리   
1              1      202112          A       골목상권   2110427    노원 을지대학교병원   
2              2      202112          A       골목상권   2110755      동일여자고등학교   
3              3      202112          A       골목상권   2110739    시흥3동 우편취급국   
4              4      202112          A       골목상권   2111017      송파나루역 3번   
...          ...         ...        ...        ...       ...           ...   
1666        1666      202112          U       관광특구   1001493  동대문패션타운 관광특구   
1667        1667      202112          U       관광특구   1001494    종로·청계 관광특구   
1668        1668      202112          U       관광특구   1001495       잠실 관광특구   
1669        1669      202112          U       관광특구   1001496   강남 마이스 관광특구   
1670        1670      202112          U       관광특구   1001491      이태원 관광특구   

      XCNTS_VALU  YDNTS_VALU  SIGNGU_CD  ADSTRD_CD  \
0         191934      438813      11545   11545710   
1         206265      459431      11350   11350611   
2         192300      439070      11545   11545680   
3         191289      438014      11545   11545690   
4         210222      445516      11710   11710562   
...          ...         ...        ...        ...   
1666      200974      451975      11140   11140590   
1667      199768      452273      11110   11110615   
1668      210191      446339      11710   11710566   
1669      205310      445727      11680   11680580   
1670      199503      448326      11170   11170650   

                                               geometry  
0     192008.56 438553.17, 191987.74 438544.51, 1919...  
1     206296.88 459321.71, 206238.71 459299.03, 2062...  
2     192443.04 439028.23, 192507.94 438996.04000000...  
3     191390.84 437878.79000000004, 191371.65 437875...  
4     210354.67 445450.29000000004, 210314.56 445371...  
...                                                 ...  
1666  201553.986325298 452217.686202384, 201565.1128...  
1667  198035.74942062798 452283.91062634904, 198019....  
1668  208228.91417088 445674.946959898, 208197.64944...  
1669  205328.897308437 446048.922964627, 205339.5607...  
1670  199226.204779015 447867.577119885, 199222.7606...  

[1671 rows x 11 columns]

In [ ]:
import math

In [ ]:
def trdar(i):
  try:
    x=main_data['좌표정보(X)'].astype(float)
    y=main_data['좌표정보(Y)'].astype(float)
    return np.where(x_std**2)>=(상권['XCNTS_VALU'][i]-x)**2+(상권['YDNTS_VALU'][i]-y)**2
  except:
    pass

In [ ]:
rang['상권'] = rang['num'].map(lambda x : trdar(x))

In [ ]:
1#빈 리스트에 모두 추가하는 방식
f_list=[]
for i in range(len(rang)):
  try:
    for j in range(len(rang['상권'].iloc[i][0])):
       f_list.append(rang['상권'].iloc[i][0][j])
  except:
    pass  

In [ ]:
len(f_list)

0

In [ ]:
len(set(f_list))

0

In [ ]:
x=main_data['좌표정보(X)'].astype(float)
y=main_data['좌표정보(Y)'].astype(float)

In [ ]:
np.where(x_std**2)>=(상권['XCNTS_VALU'][i]-x)**2+(상권['YDNTS_VALU'][i]-y)**2

ValueError: ignored

In [ ]:
상권['XCNTS_VALU']

In [ ]:
x